community/index/src/test/java/org/neo4j/index/internal/gbptree/GBPTreeTest.java

```diff
@@ -436,6 +449,66 @@ public void shouldAllowClosingWriterMultipleTimes() throws Exception
         // THEN that should be OK
     }
 
+    @Test
```

Comment:
Is it in the contract that close should always perform a checkpoint? Otherwise we should also test close with header here, if implementation change in future.